pkgs/applications/gis/qgis/wrapped.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv, lib, fetchurl, openssl, python3Packages, makeWrapper, symlinkJoin
+, qgis-unwrapped
+}:
+with lib;
+symlinkJoin rec {
+  inherit (qgis-unwrapped) version;
+  name = "qgis-${version}";
+
+  paths = [ qgis-unwrapped ];
+
+  nativeBuildInputs = [ makeWrapper python3Packages.wrapPython ];
+
+  pythonInputs = qgis-unwrapped.pythonBuildInputs ++
+                 (with python3Packages; [ chardet dateutil pyyaml pytz requests urllib3 ] );
+
+  # use the source archive directly to avoid rebuilding when changing qgis distro
+  inherit (qgis-unwrapped) src;
+
+  postBuild = ''
+    unpackPhase
+
+    buildPythonPath "$pythonInputs"
+
+    wrapProgram $out/bin/qgis \
+      --prefix PATH : $program_PATH \
+      --prefix PYTHONPATH : $program_PYTHONPATH \
+      --prefix LD_LIBRARY_PATH : ${stdenv.lib.makeLibraryPath [ openssl ]}
```

Comment:
this should be patched instead.

Reply:
This is probably not necessary anymore. `ldd $(nix-build -A qgis)/bin/.qgis-wrapped | grep openssl` shows it is linked correctly.